element-connector/src/main/java/org/eclipse/californium/elements/EndpointContextResolver.java

```diff
@@ -0,0 +1,38 @@
+/*******************************************************************************
+ * Copyright (c) 2019 Bosch Software Innovations GmbH and others.
+ * 
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * and Eclipse Distribution License v1.0 which accompany this distribution.
+ * 
+ * The Eclipse Public License is available at
+ *    http://www.eclipse.org/legal/epl-v10.html
+ * and the Eclipse Distribution License is available at
+ *    http://www.eclipse.org/org/documents/edl-v10.html.
+ * 
+ * Contributors:
+ *    Bosch Software Innovations GmbH - initial implementation
+ ******************************************************************************/
+package org.eclipse.californium.elements;
+
+/**
+ * Interface for resolving the endpoint identity from a endpoint context.
+ */
+public interface EndpointContextResolver {
```

Comment:
Should we choose a name which express that this class is responsible to extract an identifier ?

Reply:
Make a proposal.

Reply:
EndpointIdentifierExtractor ?
EndpointIdentifierFactory ?

Reply:
With resolver I associated, that what ever is used to convert the endpoint context into the "identifier" could be applied. Potentially a micro-service or DNS lookup. 
An extractor would something, which takes a part, a factory, which creates something out of the available.
So for me, resolver comes closest to my idea.

Reply:
In fact this is not really resolver which disturb me but the fact there is not `Identifier` in the class name.

Reply:
EndpointIdentifierResolver?

Reply:
EndpointIdentityResolver?

Reply:
both are OK for me :)